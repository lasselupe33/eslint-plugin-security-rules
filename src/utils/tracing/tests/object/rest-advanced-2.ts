type MyObj = {
  name: string;
  animals: Record<string, { name: string; age: number }>;
};

let object: MyObj = {
  name: "Ball",
  animals: {
    cat: {
      name: "Mittens",
      age: 3,
    },
    dog: {
      name: "Fido",
      age: 2,
    },
  },
};

if (Math.random() > 0.5) {
  object = {
    name: "Derp",
    animals: {
      ...object.animals,
      turtle: {
        name: "Turtis",
        age: 1,
      },
    },
  };
}

const totalAge =
  (object.animals["cat"]?.age ?? 0) +
  (object.animals["dog"]?.age ?? 0) +
  (object.animals["turtle"]?.age ?? 0);

const start = totalAge;

export {};
