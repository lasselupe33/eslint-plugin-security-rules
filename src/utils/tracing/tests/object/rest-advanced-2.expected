3
3
30
2
2
20
__unable to follow object expression__
10
